modules/mono/csharp_script.cpp

```diff
@@ -788,11 +788,7 @@ bool CSharpLanguage::is_assembly_reloading_needed() {
 
 	GDMonoAssembly *proj_assembly = gdmono->get_project_assembly();
 
-	String appname = ProjectSettings::get_singleton()->get("application/config/name");
-	String appname_safe = OS::get_singleton()->get_safe_dir_name(appname);
-	if (appname_safe.is_empty()) {
-		appname_safe = "UnnamedProject";
-	}
+	String appname_safe = ProjectSettings::get_singleton()->get_safe_project_name();
```

Comment:
That's a good chance but I'd have left it out of this PR to focus on the save fix (my mistake for commenting in the wrong context). I can see more places which currently rely on `get_safe_dir_name` to sanitize `application/config/name` - e.g. `OS_Unix::get_user_data_dir()` - so it would be worth refactoring them all in a dedicated PR).

Reply:
But OK it's probably late for you on Thanksgiving's eve, so we can also merge this as is and do further tweaks (such as exposing this to scripting, and using in more places) in a follow-up PR later.